common4j/src/main/com/microsoft/identity/common/java/authorities/Authority.java

```diff
@@ -157,9 +168,7 @@ public static Authority getAuthorityFromAuthorityUrl(String authorityUrl) {
             } else if (CIAM.equalsIgnoreCase(authorityTypeStr)) {
                 authority = new CIAMAuthority(authorityUrl);
             } else if (AAD_NA.equalsIgnoreCase(authorityTypeStr) && configuredAuthority instanceof NativeAuthCIAMAuthority) {
-                authority = new NativeAuthCIAMAuthority(
-                        authorityUrl,
-                        ((NativeAuthCIAMAuthority) configuredAuthority).getClientId());
+                authority = new NativeAuthCIAMAuthority(authorityUrl, clientId);
```

Comment:
Can we do a client ID null check here? If it's AAD_NA && it's a NativeAuthCIAMAuthority, we check if the client ID is null. If so, throw an exception (or do whatever else is appropriate, given the existing logic). We shouldn't be composing NativeAuthCIAMAuthority with a null client ID.

Reply:
Done.